### PR TITLE
Adding group filter to user management/proj list

### PR DESCRIPTION
**Pull request by jdtoombs**
New grouped filter now added to users section and project list view
**Update: now with frontend unit tests**

### CODE REVIEW

File: frontend/src/features/admin/users/ManageUsers.test.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ describe('Manage Users Component', () => {
     expect(table.props.data.length).toBe(2);
   });
```
Comment (jdtoombs) [inline frontend/src/features/admin/users/ManageUsers.test.tsx:109]: will come back to this - figuring out what's going on

**Review by Fosol — APPROVED** [review]

File: frontend/src/utils/YupSchema.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export const AgencyEditSchema = Yup.object().shape({
   name: Yup.string()
     .max(100, 'Agency name must be less than 100 characters')
     .required('An agency name is required.'),
-  code: Yup.string().required('An agnecy code is required.'),
```
Comment (jdtoombs) [inline frontend/src/utils/YupSchema.ts:50]: Typo from demo

File: frontend/src/components/SearchBar/FilterBar.tsx [hunk 3]
```diff
@@ -77,8 +77,8 @@ const FilterBar = <T extends object = {}>(props: PropsWithChildren<IProps<T>>) =
                 <Col className="bar-item flex-grow-0 plus-button">
                   <PlusButton
```
Comment (jdtoombs) [inline frontend/src/components/SearchBar/FilterBar.tsx:78]: was causing an ugly warning message so had to change the name

File: frontend/src/features/admin/edit-user/EditUserPage.test.tsx [hunk 4]
```diff
@@ -1,8 +1,6 @@
 import EditUserPage from './EditUserPage';
 import React from 'react';
 import renderer, { act } from 'react-test-renderer';
-import Adapter from 'enzyme-adapter-react-16';
```
Comment (jdtoombs) [inline frontend/src/features/admin/edit-user/EditUserPage.test.tsx:4]: No need for Enzyme anymore

File: frontend/src/features/admin/users/ManageUsers.test.tsx [hunk 5]
```diff
@@ -1,8 +1,6 @@
 import React from 'react';
 import { createMemoryHistory } from 'history';
 import { Router } from 'react-router-dom';
-import Adapter from 'enzyme-adapter-react-16';
```
Comment (jdtoombs) [inline frontend/src/features/admin/users/ManageUsers.test.tsx:4]: No need for Enzyme here anymore as well

File: frontend/src/features/admin/users/ManageUsers.test.tsx [hunk 6]
```diff
@@ -107,11 +103,6 @@ describe('Manage Users Component', () => {
     expect(table.props.data.length).toBe(2);
   });
 
-  it('displays enabled agencies', () => {
```
Comment (jdtoombs) [inline frontend/src/features/admin/users/ManageUsers.test.tsx:110]: No longer a `select `option, renders as an `input`. This functionality will be tested by the unit test written for `ParentGroupedFilter.tsx`

File: frontend/src/features/admin/users/ManageUsers.test.tsx [hunk 7]
```diff
@@ -107,9 +103,18 @@ describe('Manage Users Component', () => {
...
-    const { queryByText } = testRender(getStore());
-    expect(queryByText('agencyVal')).toBeVisible();
+  it('displays enabled agencies via autocomplete', async () => {
+    const { getAllByRole, container } = testRender(getStore());
+    const agency = container.querySelector('input[name="agency"]');
+
+    await wait(() => {
```
Comment (jdtoombs) [inline frontend/src/features/admin/users/ManageUsers.test.tsx:110]: had to start typing for it to be visible in DOM

File: frontend/src/components/SearchBar/ParentGroupedFilter.test.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  await wait(() => {
+    fireEvent.change(test!, {
+      target: {
+        value: 'o',
+      },
+    });
+  });
+  expect(getAllByRole('option')).toHaveLength(2);
+});
+
```
Comment (jdtoombs) [inline frontend/src/components/SearchBar/ParentGroupedFilter.test.tsx:59]: Trying to figure out why these aren't working, they are in practice but not in the unit tests 😞

**Pull request merged**